deeplearning4j-zoo/src/main/java/org/deeplearning4j/zoo/model/NASNet.java

```diff
@@ -0,0 +1,193 @@
+ *
+ * @note If using the IMAGENETLARGE weights, the input shape is (3, 331, 331).
+ * @author Justin Long (crockpotveggies)
+ *
+ */
+@AllArgsConstructor
+@Builder
+public class NASNet extends ZooModel {
+
+    @Builder.Default private long seed = 1234;
+    @Builder.Default private int[] inputShape = new int[] {3, 224, 224};
+    @Builder.Default private int numClasses = 0;
+    @Builder.Default private WeightInit weightInit = WeightInit.RELU;
+    @Builder.Default private IUpdater updater = new AdaDelta();
+    @Builder.Default private CacheMode cacheMode = CacheMode.DEVICE;
+    @Builder.Default private WorkspaceMode workspaceMode = WorkspaceMode.ENABLED;
+    @Builder.Default private ConvolutionLayer.AlgoMode cudnnAlgoMode = ConvolutionLayer.AlgoMode.PREFER_FASTEST;
+
+    // NASNet specific
+    @Builder.Default private int numBlocks = 6;
+    @Builder.Default private int penultimateFilters = 1056;
+    @Builder.Default private int stemFilters = 96;
+    @Builder.Default private int filterMultiplier = 2;
+    @Builder.Default private boolean skipReduction = true;
+
+    private NASNet() {}
+
+    @Override
+    public String pretrainedUrl(PretrainedType pretrainedType) {
+        if (pretrainedType == PretrainedType.IMAGENET)
+            return "http://blob.deeplearning4j.org/models/nasnetmobile_dl4j_inference.v1.zip";
+        else if (pretrainedType == PretrainedType.IMAGENETLARGE)
+            return "http://blob.deeplearning4j.org/models/nasnetlarge_dl4j_inference.v1.zip";
+        else
+            return null;
+    }
+
+    @Override
+    public long pretrainedChecksum(PretrainedType pretrainedType) {
+        if (pretrainedType == PretrainedType.IMAGENET)
+            return 1654817155L;
+        else if (pretrainedType == PretrainedType.IMAGENETLARGE)
+            return 1654817155L;
+        else
+            return 0L;
+    }
+
+    @Override
+    public Class<? extends Model> modelType() {
+        return ComputationGraph.class;
+    }
+
+    @Override
+    public ComputationGraph init() {
+        ComputationGraphConfiguration.GraphBuilder graph = graphBuilder();
+
+        graph.addInputs("input").setInputTypes(InputType.convolutional(inputShape[2], inputShape[1], inputShape[0]));
+
+        ComputationGraphConfiguration conf = graph.build();
+        ComputationGraph model = new ComputationGraph(conf);
+        model.init();
+
+        return model;
+    }
+
+    public ComputationGraphConfiguration.GraphBuilder graphBuilder() {
+
+        int filters = (int) Math.floor(penultimateFilters / 24);
+
+        ComputationGraphConfiguration.GraphBuilder graph = new NeuralNetConfiguration.Builder().seed(seed)
+                .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
+                .updater(updater)
+                .weightInit(weightInit)
+                .dist(new NormalDistribution(0.0, 0.5))
```

Comment:
Not used unless using distribution weight init - and can't be customized even then.
I'd remove entirely.